style.md

```diff
@@ -1730,9 +1737,15 @@ make(map[T1]T2, hint)
 
 Providing a capacity hint to `make()` tries to right-size the
 map at initialization time, which reduces the need for growing
-the map and allocations as elements are added to the map. Note
-that the capacity hint is not guaranteed for maps, so adding
-elements may still allocate even if a capacity hint is provided.
+the map and allocations as elements are added to the map.
+
+Note that, unlike slices, map capacity hints do not guarantee complete,
+preemptive allocation: preallocation only occurs if the specified capacity
+indicates that additional hashmap buckets will need to be allocated internally.
+This means that map allocations may still occur when adding elements up to the
+specified capacity to the map. See [makemap()] for more information.
```

Comment:
Should we try and keep this a little more detached from the implementation details, maybe a simpler summary such as:
"map capacity hints do not guarantee complete, preemptive allocation, but are used to approximate the number of hashmap buckets required" or something along those lines?